src/common/ble_gap_impl.cpp

```diff
@@ -765,7 +773,12 @@ uint32_t sd_ble_gap_sec_params_reply(adapter_t *adapter,
     
     if (p_sec_keyset)
     {
+#if NRF_SD_BLE_API_VERSION < 4
         std::memcpy(&keyset->keyset, p_sec_keyset, sizeof(ble_gap_sec_keyset_t));
+#else
+        //TODO: copy keyset properly
```

Comment:
Do you intend to complete the keyset handling here?

Reply:
Yes, merge gone bad :confounded: Will fix.

Reply:
Fixed and tested :grin:

Reply:
:+1: